modules/agentless_scanner_all_resources/main.tf

```diff
@@ -0,0 +1,368 @@
+
+resource "aws_vpc" "agentless_vpc" {
+  cidr_block = "10.0.0.0/16"
+  tags = merge(
+    local.tags,
+    {
+      Name : local.vpc_name
+    }
+  )
+}
+
+resource "aws_subnet" "agentless_subnet" {
+  vpc_id     = aws_vpc.agentless_vpc.id
+  cidr_block = "10.0.1.0/24"
+  tags = merge(
+    local.tags,
+    {
+      Name : local.subnet_name
+    }
+  )
+}
+
+resource "aws_internet_gateway" "agentless_igw" {
+  vpc_id = aws_vpc.agentless_vpc.id
+  tags = merge(
+    local.tags,
+    {
+      Name : local.ig_name
+    }
+  )
+}
+
+resource "aws_route_table" "public_route_table" {
+  vpc_id = aws_vpc.agentless_vpc.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.agentless_igw.id
+  }
+  tags = merge(
+    local.tags,
+    {
+      Name : local.route_table_name
+    }
+  )
+}
+
+resource "aws_route_table_association" "public_route_association" {
+  subnet_id      = aws_subnet.agentless_subnet.id
+  route_table_id = aws_route_table.public_route_table.id
+}
+
+# scanner resources ----------------------------------------------
+
+
+locals {
+  scanner_instance_template_name = "${var.resources_prefix}-agentless-scanner-instance-tpl"
+  scanner_security_group_name    = "${var.resources_prefix}-agentless-scanner-sg"
+  scanner_instance_name          = "${var.resources_prefix}-agentless-scanner-instance"
+  scanner_instance_volume        = "${var.resources_prefix}-agentless-scanner-volume"
+
+  scanner_instance_size = "r5b.2xlarge"
+
+  user-data = "" // this is static and will be provided soon
+}
+
+resource "aws_security_group" "scanner_security_group" {
+  name        = local.scanner_security_group_name
+  description = "Security group for uptycs side scanning"
+  vpc_id      = aws_vpc.agentless_vpc.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    local.tags,
+    {
+      Name : local.scanner_security_group_name
+    }
+  )
+}
+
+
+
+resource "aws_launch_template" "scanner_template" {
+  name                   = local.scanner_instance_template_name
+  image_id               = local.ami_map[local.region_code]
+  instance_type          = local.scanner_instance_size
+  update_default_version = true
+  key_name               = module.key_pair.key_pair_name
+  network_interfaces {
+    associate_public_ip_address = false
+    security_groups             = [aws_security_group.scanner_security_group.id]
+  }
+
+  user_data = local.user-data
+
+  block_device_mappings {
+    device_name = "/dev/sda1"
+
+    ebs {
+      volume_size           = 60
+      volume_type           = "gp3"
+      delete_on_termination = true
+    }
+  }
+
+  iam_instance_profile {
+    name = local.scanner_instance_profile_name // resource dependancy with scanner_instance_profile handled at monitor_asg
+  }
+
+  tag_specifications {
+    resource_type = "instance"
+    tags = merge(
+      local.tags,
+      {
+        Name : local.scanner_instance_name
```

Comment:
We need more tags then this so that Dyanesh can discover these.

Reply:
@usirsiwal-uptycs 
this is instance tags that we are defining in template
perhaps you are asking for template.
all resources in this terraform have tagged `"CreatedBy" = "${var.resources_prefix}-agentless"`

i will talk to dnyanesh if he needs more tagging